src/components/BulkOperations/HistoricalRecords.jsx

```diff
@@ -0,0 +1,139 @@
+import { useCallback, useState, useRef, useEffect } from 'react';
+import { Link } from 'react-router-dom';
+import classNames from 'classnames';
+import {
+  DataTable, Alert, DropdownFilter
+} from '@openedx/paragon';
+
+import Collapsible from '../Collapsible';
+import { formatDate } from '../../utils';
+import DiscoveryDataApiService from '../../data/services/DiscoveryDataApiService';
+import './BulkOperations.scss';
+
+const RECORDS_PER_PAGE = 10;
+
+const HistoricalRecords = () => {
+  const [isLoading, setIsLoading] = useState(true);
+
+  const [totalItems, setTotalItems] = useState(0);
+  const [totalPages, setTotalPages] = useState(1);
+  const [historicalRecords, setHistoricalRecords] = useState(null);
+
+  const [isError, setIsError] = useState(false);
+
+  const prevStatus = useRef();
+
+  const fetchData = useCallback((params) => {
+    let statusFilter = params.filters.find(f => f.id === 'status')?.value;
+
+    if (prevStatus.current !== statusFilter && params.pageIndex !== 0){
+        prevStatus.current = statusFilter;
+        return;
+    }
+    
+    prevStatus.current = statusFilter;
+    fetchBulkOpTasks(params.pageIndex, params.pageSize, statusFilter);
+  }, [])
+
+  async function fetchBulkOpTasks(pageNum, pageSize, statusFilter) {
+    setIsLoading(true);
+    try {
+      const response = await DiscoveryDataApiService.fetchBulkOperations(pageNum, pageSize, statusFilter);
+      console.log(response, response.data, 'DEBBBB')
+      setHistoricalRecords(response.data.results);
+      setTotalItems(response.data.count);
+      setTotalPages(Math.ceil(response.data.count / pageSize))
+    } catch (error) {
+      setIsError(true);
```

Comment:
on a successful fetch, we should set `setIsError` to false. Currently, we're not setting that anywhere.

Reply:
On a failure, the user is prompted to "Reload the page". Hence, there is no need for setting `IsError` to false.